tests/Library/Vanilla/UploadedFileTest.php

```diff
@@ -89,6 +89,23 @@ public function testPersistFileWithSpaces(string $name) {
         $this->assertStringContainsString('image-with-spaces.jpg', $file->getPersistedPath());
     }
 
+    /**
+     * Test that we can safely persist files with url_encoded characters in their name.
+     *
+     * @param string $name
+     *
+     * @dataProvider provideImagesEncodedChars
+     */
+    public function testPersistFileEncodedChars(string $name) {
+        // Perform some tests related to saving uploads.
+        $file = UploadedFile::fromRemoteResourceUrl($name);
+
+        // Save the upload.
+        $file->persistUpload();
+        $this->assertFileExists(PATH_UPLOADS.'/'.$file->getPersistedPath(), 'Final upload file is persisted');
+        $this->assertStringContainsString('my-25e5-259c-2596-25e7-2589-2587.png', $file->getPersistedPath());
```

Comment:
This test seems weirdly split. It looks like it's setup for a provider, but the assertion is hard coded for a particular expected value. If it can't be more dynamic, it probably shouldn't use a provider. However, if you can figure out how to make it _really_ work with a provider, it'll be a much more valuable test.